lib/puppet/provider/iis_pool/powershell.rb

```diff
@@ -26,45 +26,44 @@ def self.pipelines
   end
 
   def self.instances
-    inst_cmd = [
-      'Import-Module WebAdministration;',
-      '$pools=(ls "IIS:\\AppPools");',
-      '@(ForEach ($pool in $pools)',
-      '{ $p=($pool.name);',
-      'Get-ItemProperty "IIS:\\AppPools\$p" |',
-      'Select Name,State,enable32BitAppOnWin64,managedRuntimeVersion,managedPipelineMode }) |',
-      'ConvertTo-JSON'
-    ]
-    pool_names = JSON.parse(run(inst_cmd.join))
-    pool_names.collect do |pool|
-      pool_hash                 = {}
-      pool_hash[:ensure]        = pool['state'].downcase
-      pool_hash[:name]          = pool['name']
-      pool_hash[:enable_32_bit] = "#{pool['enable32BitAppOnWin64']}".to_sym || :false
-      pool_hash[:runtime]       = pool['managedRuntimeVersion']
-      pool_hash[:pipeline]      = pool['managedPipelineMode']
-      new(pool_hash)
+    inst_cmd = <<-ps1
+Import-Module WebAdministration;
+@{"Pools" = @(gci "IIS:\AppPools" | %{
```

Comment:
This is a way to always get back an array, otherwise the dynamic mechanism checks to see if single object and will assert a hash rather than array of hashes.  Tricky thing